libraries/timeslider/src/components/TimeSlider/TimeSlider.tsx

```diff
@@ -0,0 +1,370 @@
+                if (!model.inheritedWidgetConfig) {
+                    if (timeSlider.loop) {
+                        model.loop = widget.loop;
+                    }
+                    if (timeSlider.stopDelay) {
+                        model.playRate = widget.playRate;
+                    }
+                }
+            } else {
+                // Extract slider min and max range from the web map layers.
+                promises.push(updateWidgetFromLayerTimeInfos(widget, webMap));
+            }
+            // This is only asynchronous when updating the widget from layer time
+            // infos.
+            void Promise.all(promises).then(() => {
+                model.defaultStops = { ...widget.stops };
+                // Only set the remaining widget properties if the fullTimeExtent
+                // was set in the widget.
+                if (
+                    widget.fullTimeExtent &&
+                    widget.fullTimeExtent.start &&
+                    widget.fullTimeExtent.end
+                ) {
+                    if (!model.inheritedWidgetConfig) {
+                        model.inheritedWidgetConfig = true;
+                    } else {
+                        // Only override the stops only after the initial load of
+                        // the time slider.
+                        if (model.overrideStops) {
+                            widget.set("stops", {
+                                interval: new TimeInterval({
+                                    unit: model.timeIntervalUnit,
+                                    value: model.timeInterval,
+                                }),
+                            });
+                            // Ensure time slider playback will end on the latest date in the full
+                            // time extent.
+                            if (
+                                widget.effectiveStops &&
+                                widget.effectiveStops[
+                                    widget.effectiveStops.length - 1
+                                ] < widget.fullTimeExtent.end
+                            ) {
+                                widget.effectiveStops.push(
+                                    widget.fullTimeExtent.end
+                                );
+                            }
+                            // Reset the default time extent to match the new
+                            // stops.
+                            widget.set(
+                                "timeExtent",
+                                new TimeExtent({
+                                    start: widget.effectiveStops[0],
+                                    end: widget.effectiveStops[1],
+                                })
+                            );
+                        }
+                    }
+                    // Apply Time Slider model properties to the widget.
+                    widget.set("layout", model.layout);
```

Comment:
The `set()` function from the Accessor interface is occasionally useful when setting multiple things at once, or when the name of the property to set is dynamic. In this case though, the simpler and more natural way to write this would be `widget.layout = model.layout`.